opentelemetry-sdk/src/testing/logs/in_memory_exporter.rs

```diff
@@ -0,0 +1,177 @@
+///# use opentelemetry_sdk::logs::{BatchLogProcessor, LoggerProvider};
+///# use opentelemetry_sdk::runtime;
+///
+///# #[tokio::main]
+///# async fn main() {
+///    //Create an InMemoryLogsExporter
+///    let exporter: InMemoryLogsExporter = InMemoryLogsExporterBuilder::default().build();
+///    //Create a LoggerProvider and register the exporter
+///    let logger_provider = LoggerProvider::builder()
+///        .with_log_processor(BatchLogProcessor::builder(exporter.clone(), runtime::Tokio).build())
+///        .build();
+///    // Setup Log Appender for the log crate.
+///    let otel_log_appender = OpenTelemetryLogBridge::new(&logger_provider);
+///    log::set_boxed_logger(Box::new(otel_log_appender)).unwrap();
+///    log::set_max_level(Level::Info.to_level_filter());
+///    // Emit logs using macros from the log crate.
+///    error!("hello from {}. My price is {}", "apple", 2.99);
+///    warn!("warn!");
+///    info!("test log!");
+///    logger_provider.force_flush();
+///    let emitted_logs = exporter.get_emitted_logs().unwrap();
+///    for log in emitted_logs {
+///        println!("{:?}", log);
+///    }
+///# }
+///
+/// ```
+///
+#[derive(Debug, Clone)]
+pub struct InMemoryLogsExporterBuilder {}
+
+impl Default for InMemoryLogsExporterBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl InMemoryLogsExporterBuilder {
+    /// Creates a new instance of `InMemoryLogsExporter`.
+    ///
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    /// Creates a new instance of `InMemoryLogsExporter`.
+    ///
+    pub fn build(&self) -> InMemoryLogsExporter {
+        InMemoryLogsExporter {
+            logs: Arc::new(Mutex::new(Vec::new())),
+        }
+    }
+}
+
+impl InMemoryLogsExporter {
+    /// Returns the logs emitted via Logger as a vector of `LogData`.
+    ///
+    /// # Errors
+    ///
+    /// Returns a `LogError` if the internal lock cannot be acquired.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use opentelemetry_sdk::testing::logs::{InMemoryLogsExporter, InMemoryLogsExporterBuilder};
+    ///
+    /// let exporter = InMemoryLogsExporterBuilder::default().build();
+    /// let emitted_logs = exporter.get_emitted_logs().unwrap();
+    /// ```
+    ///
+    pub fn get_emitted_logs(&self) -> LogResult<Vec<LogData>> {
+        self.logs
+            .lock()
+            .map(|logs_guard| logs_guard.iter().cloned().collect())
+            .map_err(LogError::from)
+    }
+
+    /// Clears the internal (in-memory) storage of logs.
+    ///
+    /// # Errors
+    ///
+    /// Returns a `LogError` if the internal lock cannot be acquired.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use opentelemetry_sdk::testing::logs::{InMemoryLogsExporter, InMemoryLogsExporterBuilder};
+    ///
+    /// let exporter = InMemoryLogsExporterBuilder::default().build();
+    /// exporter.reset();
+    /// ```
+    ///
+    pub fn reset(&self) {
```

Comment:
The comment mentioned this function may return an error but it doesn't?

Reply:
Removed this, thanks for spotting